### PR TITLE
update dep, switch to ESM, require 12.20

### PR DESCRIPTION
**Pull request by jimmywarting**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ declare namespace callsites {
 		/**
 		Returns the current function.
 		*/
-		getFunction(): Function | undefined;
+		getFunction(): () => void;
```
Comment (sindresorhus) [inline index.d.ts:16]: I don't think this is correct.

File: index.js [hunk 2]
```diff
@@ -1,11 +1,9 @@
-'use strict';
-
 const callsites = () => {
 	const _prepareStackTrace = Error.prepareStackTrace;
 	Error.prepareStackTrace = (_, stack) => stack;
-	const stack = new Error().stack.slice(1);
+	const stack = new Error('_').stack.slice(1);
```
Comment (sindresorhus) [inline index.js]: Just use ESLint ignore comment.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (jimmywarting) [reply index.d.ts:16]: what is? I'm no ts expert, it was complaining about it...

Comment (sindresorhus) [reply index.d.ts:16]: Two things. You're removing the optionality. And you cannot know that the function returns void. You need an AnyFunction.

Comment (jimmywarting) [reply index.d.ts:16]: how do i write a AnyFunction? all i get is:

```

  index.d.ts:16:18
  ✖  16:18  Don't use Function as a type. Use a specific function type instead, like () => void.  @typescript-eslint/ban-types

  index.test-d.ts:9:13
  ✖   9:13  Don't use Function as a type. Use a specific function type instead, like () => void.  @typescript-eslint/ban-types

```

**Pull request closed without merging**